Implement D19 geometry

darominski opened the pull request:
This PR adds an Instrument Definition File for a previously unsupported instrument D19, operated at the ILL. The instrument is a single crystal diffractometer in Busing & Levy geometry.

**To test:**

<!-- Instructions for testing. -->

1. In the workbench, run `LoadEmptyInstrument` with 'D19' as the `InstrumentName`.
2. The created instrument should have an angular span of about 120 degrees in theta.
3. In the Z- Axis view, the left-most tube (Tube 1) should have theta of 46.38, while the right-most tube (Tube 640) should be at theta 73.49.
4. The distance from the sample to the detectors on the equatorial plane should be 0.76 m.

Fixes #31166 . 

Release note added.

<!-- Ensure the base of this PR is correct (e.g. release-next or master)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

darominski committed:
Added D19 IDF

darominski committed:
Added D19 to the list of ILL instruments

darominski committed:
Release note for D19 instrument geometry

DannyHindson commented:
@darominski - please can you fix the merge conflict on this PR

darominski committed:
Merge branch 'master' into 31166_d19_geometry

darominski committed:
Release note for D19 instrument IDF

DannyHindson commented:
@darominski - I'm afraid there's another merge conflict in diffraction.rst (busy day on that file!). Please can you fix again?

darominski committed:
Merge branch 'master' into 31166_d19_geometry

darominski commented:
> @darominski - I'm afraid there's another merge conflict in diffraction.rst (busy day on that file!). Please can you fix again?

It seems so! In the previous commit I moved the release note from v6.1 to v6.2, and fixed the merge conflict in the most recent one.

darominski committed:
Merge branch 'master' into 31166_d19_geometry
